pyro/poutine/reparam_messenger.py

```diff
@@ -13,6 +13,10 @@ class ReparamMessenger(Messenger):
     constructor.  See the :mod:`pyro.infer.reparam` module for available
     reparameterizers.
 
+    Note some reparameterizers can examine the ``*args,**kwargs`` inputs of
```

Comment:
The changes in this file are to allow data-dependent reparameterizers, e.g. `q(z|x)` where `x` is input as model `*args,**kwargs`.